llvm/lib/Target/AMDGPU/SIInsertHardClauses.cpp

```diff
@@ -185,11 +191,18 @@ class SIInsertHardClauses {
   }
 
   bool run(MachineFunction &MF) {
-
     ST = &MF.getSubtarget<GCNSubtarget>();
     if (!ST->hasHardClauses())
       return false;
 
+    unsigned MaxClauseLength = MF.getFunction().getFnAttributeAsParsedInteger(
+        "amdgpu-hard-clause-length-limit", HardClauseLengthLimit);
+    if (HardClauseLengthLimit.getNumOccurrences())
+      MaxClauseLength = HardClauseLengthLimit;
+    MaxClauseLength = std::min(MaxClauseLength, ST->maxHardClauseLength());
+    if (!MaxClauseLength)
```

Comment:
Maybe skip the pass if the max is 1?
```suggestion
    if (MaxClauseLength <= 1)
```

Reply:
Yes, that makes sense.